--- a/setup.py
+++ b/setup.py
@@ -358,6 +358,8 @@
         # On Debian /usr/local is always used, so we don't include it twice
         #add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
         #add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
+        add_dir_to_list(self.compiler.include_dirs,
+                        '/usr/include/%s' % os.environ['DEB_HOST_MULTIARCH'])
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
@@ -410,6 +412,8 @@
         lib_dirs = self.compiler.library_dirs + [
             '/lib64', '/usr/lib64',
             '/lib', '/usr/lib',
+            '/lib/%s' % os.environ['DEB_HOST_MULTIARCH'],
+            '/usr/lib/%s' % os.environ['DEB_HOST_MULTIARCH'],
             ]
         inc_dirs = self.compiler.include_dirs + ['/usr/include']
         exts = []
